### ISSUE TITLE
building against binutils 2.39 libbfd fails

### ISSUE THREAD

**Issue by nekopsykose** [iovisor/bpftrace#2324]
version: 0.15.0  
```
ninja: job failed: /usr/lib/ccache/bin/g++ -DBPFTRACE_VERSION=\"v20220809-120-g0b0f2c-dirty\" -DHAVE_BCC_CREATE_MAP -DHAVE_BCC_ELF_FOREACH_SYM -DHAVE_BCC_KFUNC -DHAVE_BCC_PROG_LOAD -DHAVE_BCC_PROG_LOAD_XATTR -DHAVE_BCC_USDT_ADDSEM -DHAVE_BCC_WHICH_SO -DHAVE_BFD_DISASM -DHAVE_LIBBPF -DHAVE_LIBBPF_BPF_H -DHAVE_LIBBPF_BTF_DUMP -DHAVE_LIBBPF_BTF_DUMP_EMIT_TYPE_DECL -DHAVE_LIBBPF_BTF_DUMP_NEW_V0_6_0 -DHAVE_LIBBPF_BTF_TYPE_CNT -DHAVE_LIBBPF_LINK_CREATE -DHAVE_LIBBPF_MAP_BATCH -DHAVE_LIBDW -DHAVE_NAME_TO_HANDLE_AT=1 -DLIBBCC_ATTACH_KPROBE_SIX_ARGS_SIGNATURE -DLIBBCC_ATTACH_UPROBE_SEVEN_ARGS_SIGNATURE -DLIBBFD_DISASM_FOUR_ARGS_SIGNATURE -DLLVM_ORC_V2 -DLLVM_VERSION_MAJOR=14 -DLLVM_VERSION_MINOR=0 -DLLVM_VERSION_PATCH=6 -I/home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/build -I/home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src -I/home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src/ast -isystem /usr/lib/llvm14/include -Os -fomit-frame-pointer -fdiagnostics-color=always -Os -DNDEBUG   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Wall -Wextra -Wundef -Wpointer-arith -Wcast-align -Wwrite-strings -Wcast-qual -Wunreachable-code -Wdisabled-optimization -fdiagnostics-color=always -std=c++17 -MD -MT src/CMakeFiles/runtime.dir/bfd-disasm.cpp.o -MF src/CMakeFiles/runtime.dir/bfd-disasm.cpp.o.d -o src/CMakeFiles/runtime.dir/bfd-disasm.cpp.o -c /home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src/bfd-disasm.cpp
/home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src/bfd-disasm.cpp: In function 'bpftrace::AlignState bpftrace::is_aligned_buf(void*, uint64_t, uint64_t)':
/home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src/bfd-disasm.cpp:58:24: error: too few arguments to function 'void init_disassemble_info(disassemble_info*, void*, fprintf_ftype, fprintf_styled_ftype)'
   58 |   init_disassemble_info(&info, stdout, fprintf_nop);
      |   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/demon/src/aports/community/bpftrace/src/bpftrace-0.15.0/src/bfd-disasm.cpp:14:
/usr/include/dis-asm.h:472:13: note: declared here
  472 | extern void init_disassemble_info (struct disassemble_info *dinfo, void *stream,
      |             ^~~~~~~~~~~~~~~~~~~~~
ninja: subcommand failed
```

this is because binutils 2.39 now takes some extra fprintf-styled-function argument for `init_disassemble_info`.

**Comment by viktormalik** [comment]
Hi @nekopsykose, this should be fixed by #2328, could you give it a try?

**Comment by nekopsykose** [comment]
confirmed, fixes the build :)
